v2/internal/pkg/batch/concurrent_worker.go

```diff
@@ -0,0 +1,239 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/openshift/oc-mirror/v2/internal/pkg/api/v2alpha1"
+	clog "github.com/openshift/oc-mirror/v2/internal/pkg/log"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/manifest"
+	"github.com/openshift/oc-mirror/v2/internal/pkg/mirror"
+	"github.com/vbauerster/mpb/v8"
+	"github.com/vbauerster/mpb/v8/decor"
+)
+
+func NewConcurrentBatch(log clog.PluggableLoggerInterface,
+	logsDir string,
+	mirror mirror.MirrorInterface,
+) BatchInterface {
+	copiedImages := v2alpha1.CollectorSchema{
+		AllImages: []v2alpha1.CopyImageSchema{},
+	}
+	return &ConcurrentBatch{Log: log, LogsDir: logsDir, Mirror: mirror, CopiedImages: copiedImages}
+}
+
+type ConcurrentBatch struct {
+	Log          clog.PluggableLoggerInterface
+	LogsDir      string
+	Mirror       mirror.MirrorInterface
+	Manifest     manifest.ManifestInterface
+	CopiedImages v2alpha1.CollectorSchema
+	Progress     *ProgressStruct
+}
+
+type BatchSchema struct {
+	BatchName string
+	Size      int
+	Images    v2alpha1.CollectorSchema
+}
+
+// Worker - the main batch processor
+func (o *ConcurrentBatch) Worker(ctx context.Context, collectorSchema v2alpha1.CollectorSchema, opts mirror.CopyOptions) (v2alpha1.CollectorSchema, error) {
```

Comment:
Also   noticed a long wait over here 

`
2024/06/19 09:34:23  [INFO]   : 🕵️  going to discover the necessary images...
2024/06/19 09:34:23  [INFO]   : 🔍 collecting release images...
2024/06/19 09:34:53  [INFO]   : 🔍 collecting operator images...
2024/06/19 09:38:43  [INFO]   : 🔍 collecting additional images...
`
Only after a 2 or 3 minutes does the next line/s show - did you experience the same thing ?

Reply:
That is true! 
This is related to the fact that the catalog tag (v4.x) keeps updating, and we therefore need to wait while oc-mirror pulls the catalog image again (during the collection phase, prior to filtering). 
We created [CLID-182](https://issues.redhat.com/browse/CLID-182) for this purpose.